### PR TITLE
Additional integration test checks

### PR DESCRIPTION
**Pull request by arhimondr**
## Description

- Ensure queries run by the AbstractTestQueryFramework have their final query info set and no tasks are left behind abandoned
- Limit query execution thread pool size to detect potential thread leaks (if a query scheduler gets abandoned upon query completion)

## Non-technical explanation

N/A

## Release notes

(X) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by findepi** [comment]
Will it be not flaky as `assertMemoryPoolReleased` was?

Can you stress test on CI?

**Comment by arhimondr** [comment]
@findepi I introduced these checks when working on https://github.com/trinodb/trino/pull/14205 and have done many runs. I haven't noticed any tests being flaky due to these additional checks. But there's always a chance for a rare race condition.

File: testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java [hunk 1]
```diff
@@ -178,6 +179,55 @@ private String describeMemoryPool(TestingTrinoServer coordinator, TestingTrinoSe
         return result.toString();
     }
 
+    private void checkQueryInfosFinal()
```
Comment (losipiuk) [inline testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java:182]: there may be some tests which schedule queries in the background and do not necessarily wait for their completion. But we can address that if we see this becoming flaky.

File: testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java [hunk 2]
```diff
@@ -178,6 +179,55 @@ private String describeMemoryPool(TestingTrinoServer coordinator, TestingTrinoSe
...
+                        List<TaskInfo> taskInfos = taskManager.getAllTaskInfo();
+                        for (TaskInfo taskInfo : taskInfos) {
+                            TaskId taskId = taskInfo.getTaskStatus().getTaskId();
+                            QueryId queryId = taskId.getQueryId();
+                            String query = "unknown";
+                            try {
+                                query = queryManager.getQueryInfo(queryId).getQuery();
+                            }
+                            catch (NoSuchElementException ignored) {
+                            }
```
Comment (losipiuk) [inline testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java:222]: nit: move to previous line

File: testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java [hunk 3]
```diff
@@ -178,6 +179,55 @@ private String describeMemoryPool(TestingTrinoServer coordinator, TestingTrinoSe
...
+                            if (!taskInfo.getTaskStatus().getState().isDone()) {
+                                fail(format("Task is expected to be in done state. TaskId: %s, QueryId: %s, Query: %s ", taskId, queryId, query));
```
Comment (losipiuk) [inline testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java:224]: with new java you can do:
```
                                fail("Task is expected to be in done state. TaskId: %s, QueryId: %s, Query: %s ".formatted(taskId, queryId, query));
```

which is slightly nicer imo

File: testing/trino-tests/src/test/java/io/trino/execution/TestUserImpersonationAccessControl.java [hunk 4]
```diff
@@ -95,14 +95,14 @@ private QueryError trySelectQuery(String assumedUser)
                     .build();
 
             // start query
-            StatementClient client = newStatementClient(httpClient, clientSession, "SELECT * FROM tpch.tiny.nation");
+            try (StatementClient client = newStatementClient(httpClient, clientSession, "SELECT * FROM tpch.tiny.nation")) {
```
Comment (losipiuk) [inline testing/trino-tests/src/test/java/io/trino/execution/TestUserImpersonationAccessControl.java:98]: nit: look unrelated - separate commit?

**Review by losipiuk — APPROVED** [review]

File: testing/trino-tests/src/test/java/io/trino/execution/TestUserImpersonationAccessControl.java (hunk 4 above)
Comment (arhimondr) [reply testing/trino-tests/src/test/java/io/trino/execution/TestUserImpersonationAccessControl.java:98]: If the statement is not closed the query never finishes and doesn't receive a final TaskInfo making the newly introduced check fail

File: testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java (hunk 2 above)
Comment (arhimondr) [reply testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java:222]: When I move it to the previous line it doesn't pass the style check

File: testing/trino-tests/src/test/java/io/trino/execution/TestUserImpersonationAccessControl.java (hunk 4 above)
Comment (losipiuk) [reply testing/trino-tests/src/test/java/io/trino/execution/TestUserImpersonationAccessControl.java:98]: Still should be a preparatory commit. So adding checks and making project actually pass the checks are separate commits. (fixing makes sense even if we do not have checks).

File: testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java (hunk 2 above)
Comment (losipiuk) [reply testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java:222]: Interesting. I have checkstyle set up in IntelliJ and it did not complain. 🤷

**Comment by losipiuk** [comment]
CI: https://github.com/trinodb/trino/issues/14368

File: testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java (hunk 2 above)
Comment (arhimondr) [reply testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java:222]: Yeah, my Intellij doesn't complain either (though it is trying to format it with a new line when I try to autoformat). But then the check during the build fails.
